docs/swagger-ui-bundle/src/test/scala/sttp/tapir/swagger/bundle/SwaggerInterpreterTest.scala

```diff
@@ -0,0 +1,77 @@
+package sttp.tapir.swagger.bundle
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import io.circe.generic.auto._
+import org.http4s.HttpRoutes
+import org.http4s.blaze.server.BlazeServerBuilder
+import org.http4s.server.Router
+import org.scalatest.Assertion
+import org.scalatest.funsuite.AsyncFunSuite
+import org.scalatest.matchers.should.Matchers
+import sttp.client3._
+import sttp.model.{StatusCode, Uri}
+import sttp.tapir.generic.auto._
+import sttp.tapir.json.circe.jsonBody
+import sttp.tapir.server.http4s.Http4sServerInterpreter
+import sttp.tapir._
+
+class SwaggerInterpreterTest extends AsyncFunSuite with Matchers {
+
+  case class Author(name: String)
+  case class Book(title: String, year: Int, author: Author)
+
+  val booksListing: PublicEndpoint[Unit, Unit, Vector[Book], Any] = endpoint.get
+    .in("books")
+    .in("list" / "all")
+    .out(jsonBody[Vector[Book]])
+
+  val addBook: PublicEndpoint[Book, Unit, Unit, Any] = endpoint.post
+    .in("books")
+    .in("add")
+    .in(
+      jsonBody[Book]
+        .description("The book to add")
+        .example(Book("Pride and Prejudice", 1813, Author("Jane Austen")))
+    )
+
+  def swaggerUITest(prefix: List[String], basePrefix: List[String]): IO[Assertion] = {
+    val swaggerUIRoutes: HttpRoutes[IO] =
+      Http4sServerInterpreter[IO]().toRoutes(
+        SwaggerInterpreter(prefix = prefix, basePrefix = basePrefix)
+          .fromEndpoints[IO](List(booksListing, addBook), "The tapir library", "1.0.0")
+      )
+
+    BlazeServerBuilder[IO]
+      .bindHttp(0, "localhost")
+      .withHttpApp(Router(s"/${basePrefix.mkString("/")}" -> swaggerUIRoutes).orNotFound)
+      .resource
+      .use { server =>
+        IO {
+          val port = server.address.getPort
+          val backend: SttpBackend[Identity, Any] = HttpURLConnectionBackend()
+          val docsPath = (basePrefix ++ prefix).mkString("/")
+          val resp: Response[String] = basicRequest
+            .response(asStringAlways)
+            .get(Uri.unsafeParse(s"http://localhost:$port/$docsPath"))
```

Comment:
simpler to use sttp's uri :) `uri"http://localhost:$port/${basePrefix ++ prefix}"`